lib/opFns.js

```diff
@@ -616,18 +617,22 @@ module.exports = {
       options.gasLimit.iadd(new BN(fees.callStipend.v))
     }
 
-    if (utils.isPrecompiled(toAddress)) {
-      options.compiled = true
-      options.code = runState._precompiled[toAddress.toString('hex')]
-      makeCall(runState, options, localOpts, done)
-    } else {
-      stateManager.getContractCode(toAddress, function (err, code, compiled) {
-        if (err) return done(err)
-        options.code = code
-        options.compiled = compiled
+    // load the code
+    stateManager.getAccount(toAddress, function (err, account) {
+      if (err) return done(err)
+      if (utils.isPrecompiled(toAddress)) {
+        options.compiled = true
+        options.code = runState._precompiled[toAddress.toString('hex')]
         makeCall(runState, options, localOpts, done)
-      })
-    }
+      } else {
+        stateManager.getContractCode(toAddress, function (err, code, compiled) {
+          if (err) return done(err)
+          options.compiled = compiled || false
+          options.code = code
+          makeCall(runState, options, localOpts, done)
+        })
+      }
+    })
```

Comment:
Hmm, code looks correct, not completely understand the changing semantics here, so probably for someone else to review again.